uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/API/Query/LeaveRequestSelect.php

```diff
@@ -102,6 +102,31 @@ private function addWhere(CRM_Utils_SQL_Select $customQuery) {
     if($hasUnassignedAsTrue) {
       $conditions[] = "NOT (" . implode(' AND ', $this->hasActiveLeaveManagerCondition()) . ") 
                        OR (r.is_active IS NULL AND rt.is_active IS NULL)";
+
+      $contactID = false;
+      $contactPassedAsArray = !empty($this->params['contact_id']['IN']);
+      $contactPassedAsID = isset($this->params['contact_id']) && is_numeric($this->params['contact_id']);
+
+      if($contactPassedAsArray) {
+        $contactID = $this->params['contact_id']['IN'];
+      }
+
+      if($contactPassedAsID) {
+        $contactID = [$this->params['contact_id']];
+      }
+
+      $query = "a.contact_id NOT IN(SELECT contact_id_a FROM civicrm_relationship r LEFT JOIN
+                       civicrm_relationship_type rt ON rt.id = r.relationship_type_id WHERE
+                       ". implode(' AND ', $this->hasActiveLeaveManagerCondition());
```

Comment:
This query already has another subquery and it would be good if we could avoid this new one. Have you tried to see if this was possible?